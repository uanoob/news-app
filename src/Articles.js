import React from 'react';
import Article from './Article';

const Articles = props => {
  const { articles } = props;
  const articleElements = articles.map(article => (
    <li key={article._id}>
      <Article article={article} />
    </li>
  ));
  return <ul>{articleElements}</ul>;
};

export default Articles;
