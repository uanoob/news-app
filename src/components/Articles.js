import React from 'react';
import PropTypes from 'prop-types';
import Article from '../containers/Article';

const Articles = (props) => {
  const { articles } = props;
  const articleElements = articles.map(article => (
    <li key={article._id}>
      <Article article={article} />
    </li>
  ));
  return <ul>{articleElements}</ul>;
};

Articles.defaultProps = {
  articles: [],
};

Articles.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      comments: PropTypes.arrayOf(PropTypes.string),
      date: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      _id: PropTypes.string.isRequired,
    }),
  ),
};

export default Articles;
