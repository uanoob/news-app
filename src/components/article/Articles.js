import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Article from './Article';
import { getAllArticles } from '../../store/actions';

const styles = theme => ({
  root: {
    margin: 'auto',
    marginTop: 20,
    width: '100%',
    maxWidth: 600,
    backgroundColor: theme.palette.background.paper,
  },
  nested: {
    paddingLeft: theme.spacing.unit * 4,
  },
});

class Articles extends Component {
  state = {};

  componentDidMount() {
    const { onGetAllArticles } = this.props;
    onGetAllArticles();
  }

  render() {
    const { classes, articles } = this.props;
    return articles && articles.length !== 0 ? (
      <div className={classes.root}>
        <List component="nav">
          {articles.map(article => (
            <ListItem key={article._id}>
              <Article article={article} />
            </ListItem>
          ))}
        </List>
      </div>
    ) : null;
  }
}

Articles.defaultProps = {
  articles: [],
};

Articles.propTypes = {
  classes: PropTypes.shape({
    root: PropTypes.string.isRequired,
    nested: PropTypes.string.isRequired,
  }).isRequired,
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      author_id: PropTypes.string.isRequired,
      author_name: PropTypes.string.isRequired,
      posted_at: PropTypes.string.isRequired,
    }).isRequired,
  ),
  onGetAllArticles: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  articles: state.articles.articles,
});

const mapDispatchToProps = {
  onGetAllArticles: getAllArticles,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withStyles(styles)(Articles));
