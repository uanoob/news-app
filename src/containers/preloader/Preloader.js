import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const styles = () => ({
  progress: {
    marginTop: 50,
  },
});

function CircularIndeterminate(props) {
  const { classes } = props;
  return (
    <div>
      <CircularProgress
        className={classes.progress}
        color="primary"
        size={100}
      />
    </div>
  );
}

CircularIndeterminate.propTypes = {
  classes: PropTypes.shape({
    progress: PropTypes.string.isRequired,
  }).isRequired,
};

export default withStyles(styles)(CircularIndeterminate);
