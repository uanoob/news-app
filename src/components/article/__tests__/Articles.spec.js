import React from 'react';
import { shallow } from 'enzyme';
import List from '@material-ui/core/List';
import { Articles } from '../Articles';
import Preloader from '../../../containers/preloader/Preloader';

describe('Articles', () => {
  const props = {
    classes: { root: '' },
    articles: [],
    authorId: '',
    isLoading: false,
    isLoaded: false,
    errorMsg: null,
    onGetAllArticles: () => {},
    onGetArticlesByAuthorId: () => {},
    onClearArticles: () => {},
  };

  describe('Articles initial', () => {
    const mockGetAllArticles = jest.fn();
    const nextProps = {
      ...props,
      onGetAllArticles: mockGetAllArticles,
    };
    const articles = shallow(<Articles {...nextProps} />);

    it('renders properly', () => {
      expect(articles).toMatchSnapshot();
    });
    it('dispatches the onGetAllArticles method', () => {
      expect(mockGetAllArticles).toHaveBeenCalledTimes(1);
    });
    it('not render <List />', () => {
      expect(articles.find(List)).toHaveLength(0);
    });
    it('not render <Preloader />', () => {
      expect(articles.find(Preloader)).toHaveLength(0);
    });
    it('not render errorMsg', () => {
      expect(articles.find('span')).toHaveLength(0);
    });
  });

  describe('Articles initial + authorId', () => {
    const mockGetArticlesByAuthorId = jest.fn();
    const nextProps = {
      ...props,
      authorId: '123',
      onGetArticlesByAuthorId: mockGetArticlesByAuthorId,
    };
    const articles = shallow(<Articles {...nextProps} />);
    it('renders properly', () => {
      expect(articles).toMatchSnapshot();
    });
    it('dispatches mockGetArticlesByAuthorId method', () => {
      expect(mockGetArticlesByAuthorId).toBeCalledTimes(1);
      expect(mockGetArticlesByAuthorId).toBeCalledWith(nextProps.authorId);
    });
  });

  describe('Articles isLoading', () => {
    const nextProps = {
      ...props,
      isLoading: true,
    };
    const articles = shallow(<Articles {...nextProps} />);

    it('renders properly', () => {
      expect(articles).toMatchSnapshot();
    });
    it('render Preloader', () => {
      expect(articles.find(Preloader)).toHaveLength(1);
    });
    it('not render <List />', () => {
      expect(articles.find(List)).toHaveLength(0);
    });
    it('not render errorMsg', () => {
      expect(articles.find('span')).toHaveLength(0);
    });
  });

  describe('Articles render <List />', () => {
    const nextProps = {
      ...props,
      articles: [
        {
          _id: 'PropTypes.string.isRequired',
          title: 'PropTypes.string.isRequired',
          text: 'PropTypes.string.isRequired',
          author_id: 'PropTypes.string.isRequired',
          author_name: 'PropTypes.string.isRequired',
          created_at: 'PropTypes.string.isRequired',
          updated_at: 'PropTypes.string.isRequired',
        },
      ],
      isLoading: false,
      isLoaded: true,
    };
    const articles = shallow(<Articles {...nextProps} />);

    it('renders properly', () => {
      expect(articles).toMatchSnapshot();
    });
    it('renders <List /> template', () => {
      expect(articles.find(List)).toHaveLength(1);
    });
    it('not render <Preloader />', () => {
      expect(articles.find(Preloader)).toHaveLength(0);
    });
    it('not render errorMsg', () => {
      expect(articles.find('span')).toHaveLength(0);
    });
  });

  describe('Articles with errorMsg', () => {
    const nextProps = {
      ...props,
      errorMsg: 'Something going wrong',
    };
    const articles = shallow(<Articles {...nextProps} />);

    it('renders properly', () => {
      expect(articles).toMatchSnapshot();
    });
    it('renders errorMsg', () => {
      expect(articles.find('span').text()).toEqual(nextProps.errorMsg);
    });
  });
});
