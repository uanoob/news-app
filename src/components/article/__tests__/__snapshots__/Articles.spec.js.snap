// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles ArticlesComponent Loaded & articles array empty render correctly ArticlesComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticlesComponent
    articles={Array []}
    authorId=""
    classes={
      Object {
        "root": "",
      }
    }
    errorMsg={null}
    isLoaded={true}
    isLoading={false}
    onClearArticles={[Function]}
    onGetAllArticles={[Function]}
    onGetArticlesByAuthorId={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        null,
        <span
          id="articles-empty-message"
        >
          No articles yep
        </span>,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No articles yep",
          "className": undefined,
          "id": "articles-empty-message",
        },
        "ref": null,
        "rendered": "No articles yep",
        "type": "span",
      },
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          null,
          <span
            id="articles-empty-message"
          >
            No articles yep
          </span>,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No articles yep",
            "className": undefined,
            "id": "articles-empty-message",
          },
          "ref": null,
          "rendered": "No articles yep",
          "type": "span",
        },
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Articles ArticlesComponent componentWillUnmount render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticlesComponent
    articles={Array []}
    authorId=""
    classes={
      Object {
        "root": "",
      }
    }
    errorMsg={null}
    isLoaded={false}
    isLoading={false}
    onClearArticles={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onGetAllArticles={[Function]}
    onGetArticlesByAuthorId={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        null,
        null,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      null,
      null,
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          null,
          null,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        null,
        null,
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Articles ArticlesComponent initial + authorId render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticlesComponent
    articles={Array []}
    authorId="123"
    classes={
      Object {
        "root": "",
      }
    }
    errorMsg={null}
    isLoaded={false}
    isLoading={false}
    onClearArticles={[Function]}
    onGetAllArticles={[Function]}
    onGetArticlesByAuthorId={
      [MockFunction] {
        "calls": Array [
          Array [
            "123",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        null,
        null,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      null,
      null,
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          null,
          null,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        null,
        null,
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Articles ArticlesComponent initial render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticlesComponent
    articles={Array []}
    authorId=""
    classes={
      Object {
        "root": "",
      }
    }
    errorMsg={null}
    isLoaded={false}
    isLoading={false}
    onClearArticles={[Function]}
    onGetAllArticles={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onGetArticlesByAuthorId={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        null,
        null,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      null,
      null,
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          null,
          null,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        null,
        null,
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Articles ArticlesComponent isLoading render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticlesComponent
    articles={Array []}
    authorId=""
    classes={
      Object {
        "root": "",
      }
    }
    errorMsg={null}
    isLoaded={false}
    isLoading={true}
    onClearArticles={[Function]}
    onGetAllArticles={[Function]}
    onGetArticlesByAuthorId={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        <div>
          <WithStyles(CircularIndeterminate) />
        </div>,
        null,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(CircularIndeterminate) />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      null,
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          <div>
            <WithStyles(CircularIndeterminate) />
          </div>,
          null,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(CircularIndeterminate) />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        null,
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Articles ArticlesComponent render <List /> render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticlesComponent
    articles={
      Array [
        Object {
          "_id": "PropTypes.string.isRequired",
          "author_id": "PropTypes.string.isRequired",
          "author_name": "PropTypes.string.isRequired",
          "created_at": "PropTypes.string.isRequired",
          "text": "PropTypes.string.isRequired",
          "title": "PropTypes.string.isRequired",
          "updated_at": "PropTypes.string.isRequired",
        },
      ]
    }
    authorId=""
    classes={
      Object {
        "root": "",
      }
    }
    errorMsg={null}
    isLoaded={true}
    isLoading={false}
    onClearArticles={[Function]}
    onGetAllArticles={[Function]}
    onGetArticlesByAuthorId={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className=""
        >
          <WithStyles(List)
            component="nav"
          >
            <WithStyles(ListItem)>
              <withRouter(Connect(WithStyles(ArticleComponent)))
                article={
                  Object {
                    "_id": "PropTypes.string.isRequired",
                    "author_id": "PropTypes.string.isRequired",
                    "author_name": "PropTypes.string.isRequired",
                    "created_at": "PropTypes.string.isRequired",
                    "text": "PropTypes.string.isRequired",
                    "title": "PropTypes.string.isRequired",
                    "updated_at": "PropTypes.string.isRequired",
                  }
                }
              />
            </WithStyles(ListItem)>
          </WithStyles(List)>
        </div>,
        null,
        null,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(List)
            component="nav"
          >
            <WithStyles(ListItem)>
              <withRouter(Connect(WithStyles(ArticleComponent)))
                article={
                  Object {
                    "_id": "PropTypes.string.isRequired",
                    "author_id": "PropTypes.string.isRequired",
                    "author_name": "PropTypes.string.isRequired",
                    "created_at": "PropTypes.string.isRequired",
                    "text": "PropTypes.string.isRequired",
                    "title": "PropTypes.string.isRequired",
                    "updated_at": "PropTypes.string.isRequired",
                  }
                }
              />
            </WithStyles(ListItem)>
          </WithStyles(List)>,
          "className": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(ListItem)>
                <withRouter(Connect(WithStyles(ArticleComponent)))
                  article={
                    Object {
                      "_id": "PropTypes.string.isRequired",
                      "author_id": "PropTypes.string.isRequired",
                      "author_name": "PropTypes.string.isRequired",
                      "created_at": "PropTypes.string.isRequired",
                      "text": "PropTypes.string.isRequired",
                      "title": "PropTypes.string.isRequired",
                      "updated_at": "PropTypes.string.isRequired",
                    }
                  }
                />
              </WithStyles(ListItem)>,
            ],
            "component": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "PropTypes.string.isRequired",
              "nodeType": "class",
              "props": Object {
                "children": <withRouter(Connect(WithStyles(ArticleComponent)))
                  article={
                    Object {
                      "_id": "PropTypes.string.isRequired",
                      "author_id": "PropTypes.string.isRequired",
                      "author_name": "PropTypes.string.isRequired",
                      "created_at": "PropTypes.string.isRequired",
                      "text": "PropTypes.string.isRequired",
                      "title": "PropTypes.string.isRequired",
                      "updated_at": "PropTypes.string.isRequired",
                    }
                  }
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "article": Object {
                    "_id": "PropTypes.string.isRequired",
                    "author_id": "PropTypes.string.isRequired",
                    "author_name": "PropTypes.string.isRequired",
                    "created_at": "PropTypes.string.isRequired",
                    "text": "PropTypes.string.isRequired",
                    "title": "PropTypes.string.isRequired",
                    "updated_at": "PropTypes.string.isRequired",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      null,
      null,
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className=""
          >
            <WithStyles(List)
              component="nav"
            >
              <WithStyles(ListItem)>
                <withRouter(Connect(WithStyles(ArticleComponent)))
                  article={
                    Object {
                      "_id": "PropTypes.string.isRequired",
                      "author_id": "PropTypes.string.isRequired",
                      "author_name": "PropTypes.string.isRequired",
                      "created_at": "PropTypes.string.isRequired",
                      "text": "PropTypes.string.isRequired",
                      "title": "PropTypes.string.isRequired",
                      "updated_at": "PropTypes.string.isRequired",
                    }
                  }
                />
              </WithStyles(ListItem)>
            </WithStyles(List)>
          </div>,
          null,
          null,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(List)
              component="nav"
            >
              <WithStyles(ListItem)>
                <withRouter(Connect(WithStyles(ArticleComponent)))
                  article={
                    Object {
                      "_id": "PropTypes.string.isRequired",
                      "author_id": "PropTypes.string.isRequired",
                      "author_name": "PropTypes.string.isRequired",
                      "created_at": "PropTypes.string.isRequired",
                      "text": "PropTypes.string.isRequired",
                      "title": "PropTypes.string.isRequired",
                      "updated_at": "PropTypes.string.isRequired",
                    }
                  }
                />
              </WithStyles(ListItem)>
            </WithStyles(List)>,
            "className": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(ListItem)>
                  <withRouter(Connect(WithStyles(ArticleComponent)))
                    article={
                      Object {
                        "_id": "PropTypes.string.isRequired",
                        "author_id": "PropTypes.string.isRequired",
                        "author_name": "PropTypes.string.isRequired",
                        "created_at": "PropTypes.string.isRequired",
                        "text": "PropTypes.string.isRequired",
                        "title": "PropTypes.string.isRequired",
                        "updated_at": "PropTypes.string.isRequired",
                      }
                    }
                  />
                </WithStyles(ListItem)>,
              ],
              "component": "nav",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "PropTypes.string.isRequired",
                "nodeType": "class",
                "props": Object {
                  "children": <withRouter(Connect(WithStyles(ArticleComponent)))
                    article={
                      Object {
                        "_id": "PropTypes.string.isRequired",
                        "author_id": "PropTypes.string.isRequired",
                        "author_name": "PropTypes.string.isRequired",
                        "created_at": "PropTypes.string.isRequired",
                        "text": "PropTypes.string.isRequired",
                        "title": "PropTypes.string.isRequired",
                        "updated_at": "PropTypes.string.isRequired",
                      }
                    }
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "article": Object {
                      "_id": "PropTypes.string.isRequired",
                      "author_id": "PropTypes.string.isRequired",
                      "author_name": "PropTypes.string.isRequired",
                      "created_at": "PropTypes.string.isRequired",
                      "text": "PropTypes.string.isRequired",
                      "title": "PropTypes.string.isRequired",
                      "updated_at": "PropTypes.string.isRequired",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        null,
        null,
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Articles ArticlesComponent with errorMsg render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticlesComponent
    articles={Array []}
    authorId=""
    classes={
      Object {
        "root": "",
      }
    }
    errorMsg="Something going wrong"
    isLoaded={false}
    isLoading={false}
    onClearArticles={[Function]}
    onGetAllArticles={[Function]}
    onGetArticlesByAuthorId={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        null,
        null,
        <span
          id="articles-error-message"
        >
          Something going wrong
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Something going wrong",
          "className": undefined,
          "id": "articles-error-message",
        },
        "ref": null,
        "rendered": "Something going wrong",
        "type": "span",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          null,
          null,
          <span
            id="articles-error-message"
          >
            Something going wrong
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Something going wrong",
            "className": undefined,
            "id": "articles-error-message",
          },
          "ref": null,
          "rendered": "Something going wrong",
          "type": "span",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
